cost-events-audit-api.md

```diff
@@ -0,0 +1,115 @@
+Filter query by event type. Currently, only
+
+`add`
+
+and
+
+`delete`
+
+are accepted. (more types coming soon) Also accepts comma-separated lists, like
+
+`add,delete`
+
+.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterResourceTypes" type="string" %}
+Resource type. Currently, only
+
+`deployment`
+
+ is accepted. (more types coming soon)
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterTotalCostLowerBound" type="float" %}
+Floating-point value representing the lower bound for the total event cost.
+{% endswagger-parameter %}
+
+{% swagger-parameter in="path" name="filterTotalCostUpperBound" type="float" %}
+Floating-point value representing the upper bound for the total event cost.
+{% endswagger-parameter %}
+
+{% swagger-response status="200: OK" description="" %}
+```javascript
+{
+    "code": 200,
+    "data": [
+        {
+            "eventType": "add",
+            "resourceType": "deployment",
+            "timestamp": "2023-02-28T13:58:44.552788548Z",
+            "properties": {
+                "cluster": "cluster-one",
+                "name": "sample-deployment",
+                "namespace": "kubecost"
+            },
+            "totalRequests": {
+                "cpu": "100m",
+                "memory": "32Mi"
+            },
+            "resourceCosts": {
+                "costPerCPUCoreHour": 0.031616438356164386,
+                "costPerRAMByteHour": 3.94217363775593e-12,
+                "costPerGPUHour": 0.95
+            },
+            "costPrediction": {
+                "totalMonthlyRate": 2.4045625000000004,
+                "cpuMonthlyRate": 2.3080000000000003,
+                "ramMonthlyRate": 0.0965625,
+                "gpuMonthlyRate": 0
+            }
+        },
+        ...
+    ]
+}
+```
+{% endswagger-response %}
+{% endswagger %}
+
+## Event tracking
+
+Changes at cluster level can range from actions triggered by declarative statements submitted by users (e.g. creation of a deployment) to automated actions (e.g. cluster autoscaling) or performance events. We detect changes that would have an estimated impact on the overall cluster cost using watchers on the Kubernetes API client.
+
+The watchers are tracking change events across all namespaces within the local/primary cluster. (the cluster that the instance of Kubecost you are interacting with via HTTP is running on)
+
+### Supported events
+
+> **Note**: This page will be updated as more event types are supported.
+
+The only events supported at the moment are deployment creations and deletions.
+
+## Cost prediction
+
+The resource requirements are inferred from the Kubernetes spec of the targeted workload. For deployments, we extract the replica count, as well as the CPU, memory and GPU requirements from the pod spec.
+
+These values are then supplied to the Kubecost Predict API to produce an estimated monthly rate for the workload.
```

Comment:
It might be sufficient to say "Cost implications of cluster events are handled by passing the Kubernetes spec from the change event to the Kubecost Predict API." Does that sound right to you?